python/whylogs/api/writer/whylabs.py

```diff
@@ -469,6 +469,9 @@ def write_feature_weights(self, file: FeatureWeights, **kwargs: Any) -> Tuple[bo
             Tuple with a boolean (1-success, 0-fail) and string with the request's status code.
         """
         self._feature_weights = file.to_dict()
+        # TODO if we want to use a WhyLabsWriter object, the first time I call write_feature_weights, should it really
+        # change the default dataset id? I think this could introduce unexpected behaviors, and should be treated as a
+        # stateless call
```

Comment:
It only changes it if you ask it to... I think this is only for edge cases where you want a single writer instance to write multiple datasets.

Reply:
chatting offline about this